CRM/Sqltasks/Task.php

```diff
@@ -219,6 +219,9 @@ public function store() {
     // error_log("STORE QUERY: " . $sql);
     // error_log("STORE PARAM: " . json_encode($params));
     CRM_Core_DAO::executeQuery($sql, $params);
+    if (empty($this->task_id)) {
+      $this->task_id = CRM_Core_DAO::singleValueQuery('SELECT LAST_INSERT_ID()');
```

Comment:
This usually doesn't matter because you can't add and run an SQL task all on one page load, but tests will do just that, so I'm storing the ID after `INSERT`.